tests/test_data_processor.py

```diff
@@ -369,12 +478,64 @@ def test_max_altitude(self, data_processor):
         ],
     )
     def test_calculate_rotations(self, data_packets, expected_value):
-        d = IMUDataProcessor([])
-        d.update_data(data_packets)
+        d = IMUDataProcessor()
+        d.update(data_packets)
         rotations = d._rotated_accelerations
         assert len(rotations) == 3
-        for rot, expected_val in zip(rotations, expected_value, strict=False):
-            assert rot == pytest.approx(expected_val)
+        # Rotations has 3 arrays in it corresponding to x, y, and z
+        # I just could not understand how to get it to work with zip so I did this instead lol
+        assert rotations[0][-1] == pytest.approx(expected_value[0])
+        assert rotations[1][-1] == pytest.approx(expected_value[1])
+        assert rotations[2][-1] == pytest.approx(expected_value[2])
+
+    def test_initial_orientation(self):
```

Comment:
ideally this test should also be paramterized so it covers literally all 6(?) gravity cases. The two cases you chose are sufficient so this isn't necessary but still.

Reply:
it was a pain to write these and I figured the cases covered everything, but you're probably right. I think I'll also wait on will for this one.